pkg/providers/aws_secretsmanager.go

```diff
@@ -41,65 +40,93 @@ func (a *AWSSecretsManager) Name() string {
+	kvs, err := a.getSecret(p)
 	if err != nil {
 		return nil, err
 	}
 
-	k := secret
-
-	entries := []core.EnvEntry{}
-	for k, v := range k {
+	var entries []core.EnvEntry
+	for k, v := range kvs {
 		entries = append(entries, p.FoundWithKey(k, v))
 	}
 	sort.Sort(core.EntriesByKey(entries))
 	return entries, nil
 }
 
-func (a *AWSSecretsManager) Put(p core.KeyPath, val string) error {
-	return fmt.Errorf("%v does not implement write yet", a.Name())
+func (a *AWSSecretsManager) Put(kp core.KeyPath, val string) error {
+	k := kp.EffectiveKey()
+	return a.PutMapping(kp, map[string]string{k: val})
 }
 
-func (a *AWSSecretsManager) PutMapping(p core.KeyPath, m map[string]string) error {
-	return fmt.Errorf("%v does not implement write yet", a.Name())
-}
+func (a *AWSSecretsManager) PutMapping(kp core.KeyPath, m map[string]string) error {
+	secrets, err := a.getSecret(kp)
+	if err != nil {
+		return err
+	}
+
+	secretAlreadyExist := len(secrets) != 0
 
-func (a *AWSSecretsManager) Get(p core.KeyPath) (*core.EnvEntry, error) {
-	secret, err := a.getSecret(p)
+	utils.Merge(m, secrets)
+	secretBytes, err := json.Marshal(m)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	data := secret
-	k, ok := data[p.Env]
-	if p.Field != "" {
-		k, ok = data[p.Field]
+	secretString := string(secretBytes)
+	ctx := context.Background()
+	if secretAlreadyExist {
+		// secret already exist - put new value
+		_, err = a.client.PutSecretValue(ctx, &secretsmanager.PutSecretValueInput{SecretId: &kp.Path, SecretString: &secretString})
+		return err
 	}
 
+	// create secret
+	_, err = a.client.CreateSecret(ctx, &secretsmanager.CreateSecretInput{Name: &kp.Path, SecretString: &secretString})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *AWSSecretsManager) Get(kp core.KeyPath) (*core.EnvEntry, error) {
+	kvs, err := a.getSecret(kp)
+	if err != nil {
+		return nil, err
+	}
+
+	k := kp.EffectiveKey()
+	val, ok := kvs[k]
 	if !ok {
-		ent := p.Missing()
+		ent := kp.Missing()
 		return &ent, nil
 	}
 
-	ent := p.Found(k)
+	ent := kp.Found(val)
 	return &ent, nil
 }
 
 func (a *AWSSecretsManager) getSecret(kp core.KeyPath) (map[string]string, error) {
-	res, err := a.client.GetSecretValue(context.TODO(), &secretsmanager.GetSecretValueInput{SecretId: &kp.Path})
+	res, err := a.client.GetSecretValue(context.Background(), &secretsmanager.GetSecretValueInput{SecretId: &kp.Path})
 	if err != nil {
-		return nil, err
+		var resNotFound *smtypes.ResourceNotFoundException
+		if !errors.As(err, &resNotFound) {
+			return nil, err
+		}
+
+		// doesn't exist - do not treat as an error
+		return nil, nil
 	}
 
 	if res == nil || res.SecretString == nil {
-		return nil, fmt.Errorf("data not found at '%s'", kp.Path)
+		return nil, fmt.Errorf("data not found at %q", kp.Path)
 	}
-	m := map[string]string{}
-	err = json.Unmarshal([]byte(*res.SecretString), &m)
 
+	var secret map[string]string
+	err = json.Unmarshal([]byte(*res.SecretString), &secret)
```

Comment:
does this still work if the json object happens to have a number primitive

Reply:
`func (a *AWSSecretsManager) getSecret(kp core.KeyPath) (map[string]string, error) {
	res, err := a.client.GetSecretValue(context.TODO(), &secretsmanager.GetSecretValueInput{SecretId: &kp.Path})
	if err != nil {
		return nil, err
	}

	if res == nil || res.SecretString == nil {
		return nil, fmt.Errorf("data not found at '%s'", kp.Path)
	}
	m := map[string]interface{}{}
	err = json.Unmarshal([]byte(*res.SecretString), &m)
	if err != nil {
		return nil, err
	}
	secret := map[string]string{}
	for k,v := range m {
		secretValue, ok :=v.(string)
		if !ok { continue }
		secret[k]=secretValue
	}
	return secret, nil
}`

above is how i had to unmashal data from secrets that may not have everything as strings